io/disk/vfc-tests/vfc_dlpar.py

```diff
@@ -0,0 +1,367 @@
+        if not self.server:
+            self.cancel("Managed System not got")
+        self.dic_list = []
+        cmd = 'lshwres -r virtualio --rsubtype fc --level lpar -m %s \
+               --filter "lpar_names=%s"' % (self.server, self.lpar)
+        for line in self.run_command(cmd):
+            self.vfc_dic = {}
+            for i in line.split(","):
+                if i.split("=")[0] == "slot_num":
+                    self.vfc_dic["c_slot"] = i.split("=")[-1]
+                elif i.split("=")[0] == "remote_slot_num":
+                    self.vfc_dic["r_slot"] = i.split("=")[-1]
+                elif i.split("=")[0] == "remote_lpar_name":
+                    self.vfc_dic["r_lpar"] = i.split("=")[-1]
+            self.vfc_dic["wwpn"] = self.get_wwpn(self.vfc_dic["c_slot"])
+            self.vfc_dic["drc"] = self.get_drc_name(self.vfc_dic["c_slot"])
+            self.vfc_dic["paths"] = self.get_paths(self.vfc_dic["drc"])
+            self.dic_list.append(self.vfc_dic)
+            self.log.info("individual dictionaries: %s " % self.vfc_dic)
+
+        self.log.info("complete list : %s" % self.dic_list)
+
+    def login(self, ipaddr, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        pxh = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        pxh.force_password = True
+
+        pxh.login(ipaddr, username, password)
+        pxh.sendline()
+        pxh.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        pxh.set_unique_prompt()
+        pxh.prompt(timeout=60)
+        self.pxssh = pxh
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc %s", command)
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = con.before.decode('utf-8').splitlines()
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        return output
+
+    @staticmethod
+    def get_mcp_component(component):
+        '''
+        probes IBM.MCP class for mentioned component and returns it.
+        '''
+        for line in process.system_output('lsrsrc IBM.MCP %s' % component,
+                                          ignore_status=True, shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split()[-1].strip('{}\"')
+        return ''
```

Comment:
This function can be used to find HMC IP.
So, decide if you want to use this or get that as input from yaml.

Reply:
will see, if we can.